WorkspaceServer.Tests/Kernel/CSharpReplTests.cs

```diff
@@ -0,0 +1,139 @@
+            var kernel = new CSharpRepl();
+
+            foreach (var command in commands ?? Enumerable.Empty<IKernelCommand>())
+            {
+                await kernel.SendAsync(command);
+            }
+
+            DisposeAfterTest(kernel.KernelEvents.Subscribe(KernelEvents.Add));
+
+            return kernel;
+        }
+
+        [Fact]
+        public async Task it_returns_the_result_of_a_non_null_expression()
+        {
+            var repl = await CreateKernelAsync();
+
+            await repl.SendAsync(new SubmitCode("123"));
+
+            KernelEvents.Last()
+                        .Should()
+                        .BeOfType<ValueProduced>()
+                        .Which
+                        .Value
+                        .Should()
+                        .Be(123);
+        }
+
+        [Fact]
+        public async Task it_returns_exceptions_thrown_in_user_code()
+        {
+            var repl = await CreateKernelAsync();
+
+            await repl.SendAsync(new SubmitCode("throw new System.NotImplementedException();"));
+
+            KernelEvents.Last()
+                .Should()
+                .BeOfType<CodeSubmissionEvaluationFailed>()
+                .Which
+                .Error
+                .Should()
+                .BeOfType<NotImplementedException>();
+        }
+
+        [Fact]
+        public async Task it_notifies_when_submission_is_complete()
+        {
+            var repl = await CreateKernelAsync();
+
+            await repl.SendAsync(new SubmitCode("var a ="));
+
+            await repl.SendAsync(new SubmitCode("12;"));
+
+            KernelEvents.Should()
+                .NotContain(e => e is ValueProduced);
+
+            KernelEvents
+                .Should()
+                .Contain(e => e is CodeSubmissionEvaluated);
+        }
+
+        [Fact]
+        public async Task it_notifies_when_submission_is_incomplete()
+        {
+            var repl = await CreateKernelAsync();
+
+            await repl.SendAsync(new SubmitCode("var a ="));
+
+            KernelEvents.Should()
+                .NotContain(e => e is ValueProduced);
+
+            KernelEvents.Last()
+                .Should()
+                .BeOfType<IncompleteCodeSubmissionReceived>();
+        }
+
+        [Fact]
+        public async Task it_returns_the_result_of_a_null_expression()
+        {
+            var repl = await CreateKernelAsync();
+
+            await repl.SendAsync(new SubmitCode("null"));
+
+            KernelEvents.Last()
+                        .Should()
+                        .BeOfType<ValueProduced>()
+                        .Which
+                        .Value
+                        .Should()
+                        .BeNull();
+        }
+
+        [Fact]
+        public async Task it_does_not_return_a_result_for_a_statement()
+        {
+            var repl = await CreateKernelAsync();
+
+            await repl.SendAsync(new SubmitCode("var x = 1;"));
+
+            KernelEvents
```

Comment:
Will this case not produce the "CodeSubmissionEvaluated" event similar to the test "it_notifies_when_submission_is_complete()"